src/components/Tag/Tag.js

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled, { css } from 'react-emotion';
+
+import { themePropType } from '../../util/shared-prop-types';
+import { textMega } from '../../styles/style-helpers';
+import DefaultCloseButton from '../CloseButton';
+
+const closeButtonStyles = ({ selected, theme }) => css`
+  label: tag__close-button;
+  margin-left: ${theme.spacings.kilo};
+  vertical-align: -1px;
+
+  > svg {
+    fill: ${selected && theme.colors.white};
+  }
+`;
+
+const CloseButton = styled(DefaultCloseButton)(closeButtonStyles);
+
+const tagStyles = ({ selected, theme }) => css`
+  label: tag;
+  padding: ${theme.spacings.bit} ${theme.spacings.kilo};
+  border-radius: ${theme.borderRadius.mega};
+  border: 1px solid ${theme.colors.n300};
+  background-color: ${selected ? theme.colors.b500 : theme.colors.n100};
+  color: ${selected && theme.colors.white};
+  ${textMega({ theme })};
+  cursor: default;
+`;
+
+const tagClickableStyles = ({ onClick, theme }) =>
+  onClick &&
+  css`
+    label: tag--clickable;
+    cursor: pointer;
+
+    &:hover {
+      background-color: ${theme.colors.n300};
+    }
+  `;
+
+const iconStyles = ({ selected, theme }) => css`
+  label: tag__icon;
+  vertical-align: -2px;
+  margin-right: ${theme.spacings.bit};
+
+  > svg {
+    fill: ${selected && theme.colors.white};
+  }
+`;
+
+const IconContainer = styled('span')(iconStyles);
+const TagElement = styled('span')(tagStyles, tagClickableStyles);
+
+const allowIcon = (onRemove, icon) => {
+  if (onRemove && icon) {
+    console.warn("When tag has onRemove prop it doesn't allow icon.");
```

Comment:
Hm, this doesn't feel right to me. Ideally, this should go into the PropTypes as a custom prop checker. WDYT?

Reply:
Also true! Totally forgot the custom props checker option